Resources/public/js/adapter/fancytree.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * A tree browser adapter for the Fancytree library.
+ *
+ * @author Wouter J <wouter@wouterj.nl>
+ * @see https://github.com/mar10/fancytree
+ */
+var FancytreeAdapter = function (data_url) {
+    if (!window.jQuery || !jQuery.fn.fancytree) {
+        throw 'The FancytreeAdapter requires both jQuery and the FancyTree library.';
+    }
+
+    var requestNodeToFancytreeNode = function (requestNode) {
+        var fancytreeNode = {
+            // fixme: remove ugly replacing by just not putting stuff like this in the JSON response
+            title: requestNode.data.replace(' (not editable)', ''),
+            // fixme: also put the current node name in the JSON response, not just the complete path
+            key: requestNode.attr.id.match(/\/([^\/]+)$/)[1]
+        };
+        
+        // fixme: have more descriptive JSON keys to determine if it has children
+        if (null != requestNode.state) {
+            fancytreeNode.folder = true;
```

Comment:
at least on phpcr node level, there is a hasNodes method that would provide that information.